src/commands/commerce/extension/register.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+// What: Implemented a register command which registers an apex class via EPN to a web store
+
+// Why: Store admins will now be able to run through the full workflow of registering their Apex class from their command line
+// rather than via workbench or running custom apex code
+
+// How: command triggers query to retrieve apex class -> inserts fields into RegisteredExternalServiceTable -> returns a json response of current inserted record
+
+import { flags, SfdxCommand } from '@salesforce/command';
+import { Messages, Org, SfdxError } from '@salesforce/core';
+import { forceDataRecordCreate, forceDataSoql } from '../../../lib/utils/sfdx/forceDataSoql';
+import { StatusFileManager } from '../../../lib/utils/statusFileManager';
+
+Messages.importMessagesDirectory(__dirname);
+
+const TOPIC = 'extension';
+const CMD = `commerce:${TOPIC}:register`;
+const msgs = Messages.loadMessages('@salesforce/commerce', 'store');
+
+export class RegisterExtension extends SfdxCommand {
+    public static readonly requiresUsername = true;
+    // Setting specific flags
+    public static description = msgs.getMessage('extension.register.cmdDescription');
+    public static example = [`sfdx ${CMD} --registered-extensionname --extensionpointname --apex-classname`];
+
+    public static flagsConfig = {
+        'registered-extensionname': flags.string({
+            char: 'r',
+            description: msgs.getMessage('extension.register.regExtensionNameFlagDescription'),
+        }),
+        extensionpointname: flags.string({
+            char: 'e',
+            description: msgs.getMessage('extension.register.extensionPointNameDescription'),
+        }),
+        'apex-classname': flags.string({
+            char: 'a',
+            description: msgs.getMessage('extension.register.apexClassNameFlagDescription'),
+        }),
```

Comment:
would there be any way to make extensionpointname, registered-extensionname and apex-classname required flags? 

I don't see a problem with extensionpointname and apex-classname since there are validations if they are missing. However if I don't specify registered-extensionname it will register it with undefined as the DeveloperName/MasterLabel. 

```
✗ ./bin/run commerce:extension:register -a Custom_Inventory -e CommerceDx_Inventory
Retrieving and registering Apex Class Custom_Inventory' using username: demo@1commerce.com
..........
Registered External Service Table = {
    "UniqueExtensionId": "1uuxx000000003FAAQ",
    "ApexClassId": "01pxx0000004UU8AAM",
    "RegisteredExtensionName": "undefined",
    "ExtensionPointName": "CommerceDx_Inventory",
    "ExternalServiceProviderType": "Extension"
};
```